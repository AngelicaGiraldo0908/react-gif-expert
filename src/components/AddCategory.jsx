import React, { useState } from "react";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const onSubmit = (event) => {
    //Evitar refresh
    event.preventDefault();
    //Validar que al menos hayan digitado un caracter
    if (inputValue.trim().length <= 1) return;
    //Agregar la categoria que digitaron en el input
    //setCategories((categories) => [...categories, inputValue]);
    onNewCategory(inputValue.trim());
    //Valor vac√≠o al agregar un elemento a la lista
    setInputValue("");
  };

  return (
    <form onSubmit={(event) => onSubmit(event)}>
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        onChange={(event) => onInputChange(event)}
      />
    </form>
  );
};
