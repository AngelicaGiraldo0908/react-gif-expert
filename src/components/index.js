//Archivo de barril
//Para reducir imports

export * from './AddCategory';
export * from './GifGrid';
export * from './Giftem';