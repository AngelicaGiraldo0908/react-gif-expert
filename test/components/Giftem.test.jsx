const { render, screen } = require("@testing-library/react");
import { Giftem } from "../../src/components/Giftem";

describe("Prueba de Giftem", () => {
  const title = "Conejos";
  const url = "https://conejos.com/conejo.png";

  test("Debe hacer match con el snapshot", () => {
    const { container } = render(<Giftem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Debe de mostrar la imagen con el URL y el ALT indicado", () => {
    render(<Giftem title={title} url={url} />);
    //expect(screen.getByRole('img').src).toBe(url);
    //expect(screen.getByRole('img').alt).toBe(title);
    const {src, alt} =screen.getByRole('img');
    expect(src).toBe(url);
    expect(alt).toBe(title);

  });

  test('Debe de mostrar el titulo en el componente', () => { 
    render(<Giftem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
   });

});
