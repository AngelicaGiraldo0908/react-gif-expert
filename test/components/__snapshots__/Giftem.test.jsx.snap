// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de Giftem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Conejos"
      src="https://conejos.com/conejo.png"
    />
    <p>
      Conejos
    </p>
  </div>
</div>
`;
